cluster-autoscaler/cloudprovider/azure/azure_scale_set.go

```diff
@@ -678,41 +777,19 @@ func addInstanceToCache(instances *[]cloudprovider.Instance, id *string, provisi
 	}
 
 	*instances = append(*instances, cloudprovider.Instance{
-		Id:     "azure://" + resourceID,
+		Id:     azurePrefix + resourceID,
 		Status: instanceStatusFromProvisioningStateAndPowerState(resourceID, provisioningState, powerState),
 	})
 }
 
-func (scaleSet *ScaleSet) getInstanceByProviderID(providerID string) (cloudprovider.Instance, bool) {
-	scaleSet.instanceMutex.Lock()
-	defer scaleSet.instanceMutex.Unlock()
-	for _, instance := range scaleSet.instanceCache {
-		if instance.Id == providerID {
-			return instance, true
-		}
-	}
-	return cloudprovider.Instance{}, false
-}
-
-func (scaleSet *ScaleSet) setInstanceStatusByProviderID(providerID string, status cloudprovider.InstanceStatus) {
```

Comment:
Moved to azure_scale_set_instance_cache.go, with an addition if scaleSet.validateInstanceCacheWithoutLock().